tools/state-viewer/src/latest_witnesses.rs

```diff
@@ -38,22 +66,86 @@ impl LatestWitnessesCmd {
             .get_latest_witnesses(self.height, self.shard_id, self.epoch_id.clone())
             .unwrap();
         println!("Found {} witnesses:", witnesses.len());
+        if let LatestWitnessesMode::Binary { ref output_dir } = self.mode {
+            if !output_dir.exists() {
+                std::fs::create_dir_all(output_dir).unwrap();
+            }
+        }
+
         for (i, witness) in witnesses.iter().enumerate() {
             println!(
-                "#{} (height: {}, shard_id: {}, epoch_id: {:?}):",
+                "#{} (height: {}, shard_id: {}, epoch_id: {:?})",
                 i,
                 witness.chunk_header.height_created(),
                 witness.chunk_header.shard_id(),
                 witness.epoch_id
             );
-            if self.pretty {
-                println!("{:#?}", witness);
-            } else if self.binary {
-                println!("{:?}", borsh::to_vec(witness).unwrap());
-            } else {
-                println!("{:?}", witness);
+            match self.mode {
+                LatestWitnessesMode::Pretty => {
+                    println!("{:#?}", witness);
+                    println!("");
+                }
+                LatestWitnessesMode::Binary { ref output_dir } => {
+                    let file_name = format!(
+                        "witness_{}_{}_{}.bin",
+                        witness.chunk_header.height_created(),
+                        witness.chunk_header.shard_id(),
+                        witness.epoch_id.0
+                    );
```

Comment:
Theoretically there could be multiple saved witnesses with the same `(height, shard_id, epoch_id)`. Let's maybe add the index `i` as well?